modules/eks/cluster/README.md

```diff
@@ -1,24 +1,133 @@
+        allowed_security_groups: []
+        enabled_cluster_log_types:
+          # Caution: enabling `api` log events may lead to a substantial increase in Cloudwatch Logs expenses.
+          - api  
+          - audit
+          - authenticator
+          - controllerManager
+          - scheduler
         oidc_provider_enabled: true
-        public_access_cidrs: ["72.107.0.0/24"]
+
+        # Allows GitHub OIDC role
+        github_actions_iam_role_enabled: true
+        github_actions_iam_role_attributes: [ "eks" ]
+        github_actions_allowed_repos:
+          - acme/infra
+
+        # We use karpenter to provision nodes
+        # See below for using node_groups
+        managed_node_groups_enabled: false
+        node_groups: {}   
+
+        # EKS IAM Authentication settings
+        # By default, you can authenticate to EKS cluster only by assuming the role that created the cluster.
+        # After the Auth Config Map is applied, the other IAM roles in
+        # `primary_iam_roles`, `delegated_iam_roles`, and `sso_iam_roles` will be able to authenticate.
+        apply_config_map_aws_auth: true
+        availability_zone_abbreviation_type: fixed
+        cluster_private_subnets_only: true
+        cluster_encryption_config_enabled: true
+        cluster_endpoint_private_access: true
+        cluster_endpoint_public_access: false
+        cluster_log_retention_period: 90
+        # Roles from the primary account to allow access to the cluster
+        # See `aws-teams` component.
+        primary_iam_roles:
+          - groups:
+              - system:masters
+              - idp:ops
+            role: devops
+          - groups:
+              - system:masters
+            role: spacelift
+        # Roles from the account owning the cluster to allow access to the cluster
+        # See `aws-team-roles` component.
+        delegated_iam_roles:
+          - groups:
+              - system:masters
+              - idp:ops
+            role: admin
+          - groups:
+              - idp:poweruser
+            role: poweruser
+          - groups:
+              - idp:observer
+            role: observer
+          - groups:
+              - system:masters
+            role: terraform
+        # Roles from AWS SSO allowing cluster access
+        # See `aws-sso` component.
+        sso_iam_roles:
+          - groups:
+              - idp:observer
+              - idp:observer-extra
+            role: ReadOnlyAccess
+          - groups:
+              - system:masters
+              - idp:ops
+            role: AdministratorAccess
+        fargate_profiles:
+          karpenter:
+            kubernetes_namespace: karpenter
+            kubernetes_labels: null
+        karpenter_iam_role_enabled: true
+```
+
+### Usage with Node Groups
+
+The `eks/cluster` component also supports node groups! In order to add a set list of nodes to 
+provision with the cluster, add values for `var.managed_node_groups_enabled` and `var.node_groups`.
```

Comment:
Can node groups and karpenter be deployed at the same time? If so, please mention that.

Reply:
added note